frmts/vrt/vrtexpression.h

```diff
@@ -0,0 +1,45 @@
+/******************************************************************************
+ *
+ * Project:  Virtual GDAL Datasets
+ * Purpose:  Implementation of GDALExpressionEvaluator.
+ * Author:   Daniel Baston
+ *
+ ******************************************************************************
+ * Copyright (c) 2024, ISciences LLC
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#pragma once
+
+#include "cpl_error.h"
+
+#include <string_view>
+#include <vector>
+
+/**
+ * Class to support evaluation of an expression using the exprtk library.
+ */
+class GDALExpressionEvaluator
```

Comment:
Maybe instead `gdal::ExprtkEvaluator` or `gdal::vrt::ExprtkExpression` ?

Reply:
> `gdal::ExprtkEvaluator`

would be fine to me as there's nothing vrt specific in it as far as I can see and we could potentially moved it upwards in the hierarchy if needed